### PR TITLE
Update ports

### PR DESCRIPTION
**Pull request by billingsley-john**
## Proposed changes

PR to update ports workplane and rotation_axis for correct azimuthal placement. As detailed in #493 

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [x] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Shimwell** [comment]
Would it be possible to manually test the vacuum vessel example and post an updated image for the docs. Much appreciated

https://github.com/ukaea/paramak/blob/develop/examples/example_parametric_components/make_vacuum_vessel_with_ports.py

**Comment by billingsley-john** [comment]
Start radius parameter needs to be added. Originally, it was thought that specifying self.point = [(start_radius, z_pos)] would work, however, this only translates the face points in the plane of the face. Alternatively, a cutting solid could be created which cuts the ports up to the specified start radius. This would be a similar implementation to the cutting wedge used for extrude shapes when rotation_angle is specified.

**Comment by billingsley-john** [comment]
Mathcha images and more thorough testing needs to be added

**Comment by billingsley-john** [comment]
PortCutterRectangular currently has an issue found by @RemDelaporteMathurin. When parameters are default, i.e workplane = "ZY", rotation_axis = "Z", produces image like below. This is strange as this does not happen for PortCutterCircular. Perhaps it could be the different implementations of the .workplane(offset) or related to rotation_axis.
Screenshot from 2020-11-20 17-25-19

**Comment by Shimwell** [comment]
image for the docs 
Screenshot from 2020-11-20 17-31-37

**Comment by Shimwell** [comment]
another image for the docs

Screenshot from 2020-11-20 17-37-12

**Comment by Shimwell** [comment]
Looks like the failing test is just due to a small volume difference between the exspected value and the obtained value.

```python
 def test_ToroidalFieldCoilTripleArc_creation_no_inner_leg(self):
        """creates a tf coil with no inner leg using the ToroidalFieldCoilRectangle
        parametric component and checks that a cadquery solid is created"""
    
        test_shape_1 = paramak.ToroidalFieldCoilTripleArc(
            R1=100, h=100, radii=(100, 200), coverages=(10, 60), thickness=10,
            distance=50, number_of_coils=1, vertical_displacement=10,
            with_inner_leg=True
        )
        test_volume_1 = test_shape_1.volume
    
        test_inner_leg = paramak.ExtrudeStraightShape(
            points=test_shape_1.inner_leg_connection_points, distance=50
        )
        inner_leg_volume = test_inner_leg.volume
    
        test_shape_2 = paramak.ToroidalFieldCoilTripleArc(
            R1=100, h=100, radii=(100, 200), coverages=(10, 60), thickness=10,
            distance=50, number_of_coils=1, vertical_displacement=10,
            with_inner_leg=False
        )
        assert test_shape_2.solid is not None
>       assert test_shape_2.volume == pytest.approx(
            test_volume_1 - inner_leg_volume)
E       AssertionError: assert 692992.5227660659 == 693136.0970425373 ± 6.9e-01
E         +692992.5227660659
E         -693136.0970425373 ± 6.9e-01

```

**Comment by RemDelaporteMathurin** [comment]
> Looks like the failing test is just due to a small volume difference between the exspected value and the obtained value.
> 
> ```python
>  def test_ToroidalFieldCoilTripleArc_creation_no_inner_leg(self):
>         """creates a tf coil with no inner leg using the ToroidalFieldCoilRectangle
>         parametric component and checks that a cadquery solid is created"""
>     
>         test_shape_1 = paramak.ToroidalFieldCoilTripleArc(
>             R1=100, h=100, radii=(100, 200), coverages=(10, 60), thickness=10,
>             distance=50, number_of_coils=1, vertical_displacement=10,
>             with_inner_leg=True
>         )
>         test_volume_1 = test_shape_1.volume
>     
>         test_inner_leg = paramak.ExtrudeStraightShape(
>             points=test_shape_1.inner_leg_connection_points, distance=50
>         )
>         inner_leg_volume = test_inner_leg.volume
>     
>         test_shape_2 = paramak.ToroidalFieldCoilTripleArc(
>             R1=100, h=100, radii=(100, 200), coverages=(10, 60), thickness=10,
>             distance=50, number_of_coils=1, vertical_displacement=10,
>             with_inner_leg=False
>         )
>         assert test_shape_2.solid is not None
> >       assert test_shape_2.volume == pytest.approx(
>             test_volume_1 - inner_leg_volume)
> E       AssertionError: assert 692992.5227660659 == 693136.0970425373 ± 6.9e-01
> E         +692992.5227660659
> E         -693136.0970425373 ± 6.9e-01
> ```

Would a rebase to develop fix this ?

File: paramak/parametric_shapes/extruded_circle_shape.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def __init__(
         self,
         distance,
         radius,
+        extrusion_start_offset=1,
```
Comment (RemDelaporteMathurin) [inline paramak/parametric_shapes/extruded_circle_shape.py:26]:
```suggestion
        extrusion_start_offset=0.0,
```

File: paramak/parametric_shapes/extruded_mixed_shape.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def __init__(
         distance,
         extrude_both=True,
         rotation_angle=360.0,
+        extrusion_start_offset=1,
```
Comment (RemDelaporteMathurin) [inline paramak/parametric_shapes/extruded_mixed_shape.py:28]:
```suggestion
        extrusion_start_offset=0.0,
```

Comment (Shimwell) [reply paramak/parametric_shapes/extruded_mixed_shape.py:28]: I think this is the best options

File: paramak/parametric_shapes/extruded_circle_shape.py (hunk 1 above)
Comment (Shimwell) [reply paramak/parametric_shapes/extruded_circle_shape.py]: I think this is the best options

**Comment by Shimwell** [comment]
Do we need extrusion_start_offset=1 setting in the cutters

File: paramak/parametric_components/port_cutters_circular.py [hunk 3]
```diff
@@ -24,6 +24,7 @@ def __init__(
         distance,
         workplane="ZY",
         rotation_axis="Z",
+        extrusion_start_offset=1,
```
Comment (RemDelaporteMathurin) [inline paramak/parametric_components/port_cutters_circular.py:27]: should appear in the docstrings

File: paramak/parametric_components/port_cutters_rectangular.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ def __init__(
         distance,
         workplane="ZY",
         rotation_axis="Z",
+        extrusion_start_offset=1,
```
Comment (RemDelaporteMathurin) [inline paramak/parametric_components/port_cutters_rectangular.py:31]: should appear in the docstrings

File: paramak/parametric_components/port_cutters_circular.py [hunk 5]
```diff
@@ -15,20 +15,27 @@ class PortCutterCircular(ExtrudeCircleShape):
         stl_filename (str, optional): defaults to "PortCutterCircular.stl".
         name (str, optional): defaults to "circular_port_cutter".
         material_tag (str, optional): defaults to "circular_port_cutter_mat".
+        extrusion_start_offset (float, optional): the distance between 0 and the start of the extrusion. Defaults to 1.
```
Comment (Shimwell) [inline paramak/parametric_components/port_cutters_circular.py:18]:
```suggestion
        extrusion_start_offset (float, optional): the distance between 0 and
            the start of the extrusion. Defaults to 1.
```

File: paramak/parametric_components/port_cutters_circular.py [hunk 6]
```diff
@@ -15,20 +15,28 @@ class PortCutterCircular(ExtrudeCircleShape):
         stl_filename (str, optional): defaults to "PortCutterCircular.stl".
         name (str, optional): defaults to "circular_port_cutter".
         material_tag (str, optional): defaults to "circular_port_cutter_mat".
+        extrusion_start_offset (float, optional): the distance between 0 and
+            the start of the extrusion. Defaults to 1.
```
Comment (Shimwell) [inline paramak/parametric_components/port_cutters_circular.py:19]:
```suggestion
            the start of the extrusion. Defaults to 1..
```

**Pull request merged**